contracts/GeneralizedSwapMigrator.sol

```diff
@@ -66,27 +66,24 @@ contract GeneralizedSwapMigrator is Ownable, BaseBoringBatchable {
         );
 
         for (uint8 i = 0; i < 32; i++) {
-            address oldPoolUnderlyingToken;
-            try ISwap(oldPoolAddress).getToken(i) returns (
-                IERC20 underlyingToken
-            ) {
-                oldPoolUnderlyingToken = address(underlyingToken);
+            address oldPoolToken;
+            try ISwap(oldPoolAddress).getToken(i) returns (IERC20 token) {
+                oldPoolToken = address(token);
             } catch {
                 require(i > 0, "Failed to get tokens underlying Saddle pool.");
-                oldPoolUnderlyingToken = address(0);
+                oldPoolToken = address(0);
             }
 
-            try ISwap(mData.newPoolAddress).getToken(i) returns (
-                IERC20 underlyingToken
-            ) {
+            try ISwap(mData.newPoolAddress).getToken(i) returns (IERC20 token) {
                 require(
-                    oldPoolUnderlyingToken == address(underlyingToken),
+                    oldPoolToken == address(token) &&
+                        oldPoolToken == address(mData.tokens[i]),
                     "Failed to match tokens list"
                 );
             } catch {
                 require(i > 0, "Failed to get tokens underlying Saddle pool.");
                 require(
-                    oldPoolUnderlyingToken == address(0),
+                    oldPoolToken == address(0) && i == mData.tokens.length,
```

Comment:
nit: it's generally discouraged to use double equal assertion in JS.

https://codeahoy.com/javascript/2019/10/12/==-vs-===-in-javascript/